nil-docs/zkllvm/getting-started/compiling-a-circuit.mdx

```diff
@@ -0,0 +1,193 @@
+    direction LR
+    CLANG
+    RUSTCOMP
+    IR
+    end
+    subgraph Assigner
+    direction LR
+    INPUTS
+    ASSIGNER
+    ASSIGNERRESULT
+    end
+    subgraph Transpiler
+    direction LR
+    TRANSPILER
+    PROOFBIN
+    SOLIDITY
+    end
+```
+
+Complete the below steps to follow this flow.
+
+### Running the compiler
+
+To generate the circuit IR, do the following.
+
+<Tabs groupId='language'>
+  <TabItem value='cpp' label='C++'>
+
+    For a pre-built binary installation of `clang`:
+
+    ```bash
+    clang-zkllvm -S -emit-llvm  path/to/circuit.cpp
+    ```
+
+    For a `clang` installation from sources:
+
+    ```bash
+    $CLANG_ZKLLVM -S -emit-llvm  path/to/circuit.cpp
+    ```
+
+    , where `$CLANG_ZKLLVM` is the `clang` binary built during the installation.
+
+  </TabItem>
+
+  <TabItem value='rust' label='Rust'>
+
+    For a pre-built binary installation of `rustc`:
+
+    ```bash
+    rustc +zkllvm --target assigner-unknown-unknown --emit=llvm-ir multiplication_example.rs -C opt-level=3
+    ```
+
+    Or for a pre-built binary installation of `cargo`:
+
+    ```bash
+    cargo +zkllvm build --target assigner-unknown-unknown
+    ```
+
+    For an installation of `rustc` from sources:
+
+    ```bash
+    $RUSTC --target assigner-unknown-unknown --emit=llvm-ir multiplication_example.rs -C opt-level=3
+    ```
+
+    Or for an installation of `cargo` and `rustc` from sources:
+
+    ```bash
+    RUSTC=$RUSTC $CARGO build --target assigner-unknown-unknown --release
+    ```
+
+    , where `$RUSTC` and `$CARGO` are binaries built during [**installation**](../getting-started/installation).
+
+    The `-C opt-level=3` option is needed to build the circuit into an IR that is most compatible with `assigner`. If it is ommitted, the IR will contain additional verbose code which may hinder `assigner` from generating constraints and the assignment table.
+
+  </TabItem>
+</Tabs>
+
+### Passing the IR to `assigner`
+
+At this stage, the newly generated IR and the public info file are ready to be passed to `assigner`.
+
+Run the following command to generate the circuit constraints and the assignment table.
+
+```bash
+assigner -b circuit.ll -i ../inputs/public_input.inp -c circuit.crct -t assignment.tbl -e pallas --generate-type circuit-assignment
+```
+
+### Using `transpiler` to generate the circuit proof
+
+The `.crct` and `.tbl` files can be passed to the `transpiler` tool to generate a binary with the proof and Solidity files containing the circuit.
+
+Run the below command to generate `proof.bin`.
+
+```bash
+transpiler -m gen-test-proof -c circuit.crct -t assignment.tbl -o output -e pallas
+```
+
+Execute the following command to produce Solidity files.
+
+```bash
+transpiler -m gen-evm-verifier -c circuit.crct -t assignment.tbl -o output -e pallas --optimize-gates
+```
+
+Congratulations! At this stage, we recommend taking a look at other examples in the zkLLVM repository and compiling them.
+
+## Using `zkllvm-template`
```

Comment:
Maybe we should implicitly mention somewhere, that `zkllvm-template` is only for C++ projects?

Reply:
sure, done

Reply:
Quick note: `zkllvm-template` is not going to support Rust (you say "currently supporting only C++"). Rust system just don't need a template repo, because it doesn't need precise compiler configuration in CMakeLists and etc. 

We actually have a [zkllvm-rust-template](https://github.com/NilFoundation/zkllvm-rust-template) repo, but we don't want to involve it. The idea is that user don't need to do any complex setup to be able to generate circuits. It should be his typical Rust project

Reply:
makes sense, thank you for the clarification